### PR TITLE
Right click option shouldn't show up when mouse moved after press

### PR DESCRIPTION
**Pull request by amzn-rhhong**

### CODE REVIEW

**Comment by hultonha** [comment]
@amzn-rhhong I'd recommend taking a look at a type I added called `ClickDetector`, it wraps up a lot of this logic and you can use it to ensure once a move has happened, a click event does not register. It might make the change a bit smaller (we use it with Box Select for example). It was written after `EditorContextMenu` but we should likely use it there too

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.h [hunk 1]
```diff
@@ -79,5 +81,12 @@ namespace EMStudio
         AZStd::shared_ptr<AzFramework::OrbitMotionDollyCameraInput> m_followScrollMotionCamera;
 
         AZ::Vector3 m_defaultOrbitPoint = AZ::Vector3::CreateZero();
+
+        // Properties related to the mouse event.
+        // Used to prevent right click option showing up when mouse moved between press and release.
+        int m_prevMouseX = 0.0f;
+        int m_prevMouseY = 0.0f;
```
Comment (AMZN-daimini) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.h:88]: You could use a QPoint and save some code :)

**Comment by amzn-rhhong** [comment]
@hultonha I took a look on that class. It looks like in that case I need to use the AzInput system and write a custom viewport controller just to receive the input event. Not sure if that will result in less code? For a much more complex system like camera I think that is more justified.

**Comment by amzn-rhhong** [comment]
Resolves https://github.com/o3de/o3de/issues/7757

**Review by lsemp3d — APPROVED** [review]

**Comment by hultonha** [comment]
> @hultonha I took a look on that class. It looks like in that case I need to use the AzInput system and write a custom viewport controller just to receive the input event. Not sure if that will result in less code? For a much more complex system like camera I think that is more justified.

You should not need to use the AzInput system, the type deals with its own types (you just need to map from a Qt event to `ClickDetector::ClickEvent`, so you don't have to write a custom controller or anything. You don't have to use it of course but I do think it is a pretty good fit (you just have to pass through the event and a mouse delta you track yourself). It also has quite a few tests 🙂

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.cpp [hunk 2]
```diff
@@ -291,9 +291,23 @@ namespace EMStudio
...
+    void AnimViewportWidget::mousePressEvent(QMouseEvent* event)
+    {
+        m_pixelsSinceClick = 0;
+        m_prevMousePoint = event->globalPos();
+    }
+
+    void AnimViewportWidget::mouseMoveEvent(QMouseEvent* event)
+    {
+        int deltaX = event->globalX() - m_prevMousePoint.x();
+        int deltaY = event->globalY() - m_prevMousePoint.y();
```
Comment (hultonha) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.cpp:303]: question: Could you also just use `QPoint` here and then use Manhattan distance?

**Review by hultonha — APPROVED** [review]

**Pull request merged**